lib/std/os/linux/io_uring.zig

```diff
@@ -0,0 +1,877 @@
+        }
+        return submitted;
+    }
+
+    /// Tell the kernel we have submitted SQEs and/or want to wait for CQEs.
+    /// Returns the number of SQEs submitted.
+    pub fn enter(self: *IO_Uring, to_submit: u32, min_complete: u32, flags: u32) !u32 {
+        assert(self.fd >= 0);
+        const res = linux.io_uring_enter(self.fd, to_submit, min_complete, flags, null);
+        switch (linux.getErrno(res)) {
+            0 => {},
+            // The kernel was unable to allocate memory or ran out of resources for the request.
+            // The application should wait for some completions and try again:
+            linux.EAGAIN => return error.SystemResources,
+            // The application attempted to overcommit the number of requests it can have pending.
+            // The application should wait for some completions and try again:
+            linux.EBUSY => return error.CompletionQueueOvercommitted,
+            // The SQE `fd` is invalid, or IOSQE_FIXED_FILE was set but no files were registered:
+            linux.EBADF => return error.FileDescriptorInvalid,
+            // The buffer is outside the process' accessible address space, or IORING_OP_READ_FIXED
+            // or IORING_OP_WRITE_FIXED was specified but no buffers were registered, or the range
+            // described by `addr` and `len` is not within the buffer registered at `buf_index`:
+            linux.EFAULT => return error.BufferInvalid,
+            // The SQE is invalid, or valid but the ring was setup with IORING_SETUP_IOPOLL:
+            linux.EINVAL => return error.SubmissionQueueEntryInvalid,
+            linux.ENXIO => return error.RingShuttingDown,
+            // The kernel believes our `self.fd` does not refer to an io_uring instance,
+            // or the opcode is valid but not supported by this kernel (more likely):
+            linux.EOPNOTSUPP => return error.OpcodeNotSupported,
+            // The operation was interrupted by a delivery of a signal before it could complete.
+            // This can happen while waiting for events with IORING_ENTER_GETEVENTS:
+            linux.EINTR => return error.SignalInterrupt,
+            else => |errno| return os.unexpectedErrno(errno)
+        }
+        return @truncate(u32, res);
+    }
+
+    /// Sync internal state with kernel ring state on the SQ side.
+    /// Returns the number of all pending events in the SQ ring, for the shared ring.
+    /// This return value includes previously flushed SQEs, as per liburing.
+    /// The rationale is to suggest that an io_uring_enter() call is needed rather than not.
+    /// Matches the implementation of __io_uring_flush_sq() in liburing.
+    pub fn flush_sq(self: *IO_Uring) u32 {
+        if (self.sq.sqe_head != self.sq.sqe_tail) {
+            // Fill in SQEs that we have queued up, adding them to the kernel ring.
+            const to_submit = self.sq.sqe_tail -% self.sq.sqe_head;
+            const mask = self.sq.mask.*;
```

Comment:
You should be able to cache the mask in the data structure itself (I believe rust did this? see https://github.com/stbuehler/rust-io-uring/blob/bc5fa2f67788938b9642d07fc9574d1e59163f76/io-uring/src/lib.rs#L66)

Reply:
Thanks, done!